test/cli/install/bun-add.test.ts

```diff
@@ -1351,3 +1351,61 @@ it("should add dependency without duplication (GitHub)", async () => {
   );
   await access(join(package_dir, "bun.lockb"));
 });
+
+it("should add dependencies to workspaces directly", async () => {
```

Comment:
this test is kind of confusing to me

basically it should be the equivalent of

```
bun add foo
jq package.json .dependencies.react  # should now appear
cat node_modules/react/package.json # should fail unless the workspace already had a conflicting version installed
cd ../
cat package.json # should be unmodified
```

Reply:
Oops, got distracted and ended up with a half-baked test &minus; will fix.

Reply:
Done in https://github.com/oven-sh/bun/pull/2834/commits/92f7fd35a5127f9386e08b32da2f7c5f69acfcd9